src/utils/files.ts

```diff
@@ -0,0 +1,26 @@
+import _ from "lodash"
+
+export const getFileExt = (mediaUrl: string): string | undefined => {
+  // NOTE: If it starts with data, the image is within a private repo.
+  // Hence, we will extract the portion after the specifier
+  // till the terminating semi-colon for use as the extension
+  if (mediaUrl.startsWith("data:image/")) {
+    return _.takeWhile(mediaUrl.slice(11), (char) => char !== ";").join("")
+  }
+
+  // Otherwise, this will point to a publicly accessible github url
+  return mediaUrl.split(".").pop()?.split("?").shift()
+}
+
+/**
+ * @precondition this function must only be called on a qualified filename.
+ * Ie, calling this function on a `data:...` is invalid and the function will return
+ * erroneous info.
+ *
+ * This function also assumes that the file is not nested within any directory
+ * @param name the qualified filename of the data
+ * @returns
+ */
+export const getFileName = (name: string): string | undefined => {
+  return name.split(".").at(0)
```

Comment:
actually files with 2 dots won't be allowed in our frontend because of our schema (`MediaSettingsSchema`) but i've changed this method so that it extracts the head elements of the list and joins them with a dot. (ie, `one.two.three` -> `one.two`; this is identical to how filesystem treats extensions) wdyt about this?

Reply:
am ok with this!